spec/lib/texd/cache_spec.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+RSpec.describe Texd::Cache do
+  subject(:cache) { Texd::Cache.new(3) }
+
+  it "empty cache" do
+    expect(cache.count).to eq 0
+    expect(cache.read(:one)).to eq nil
+  end
+
+  it "filling cache" do
+    cache.write(:one, "1")
+    expect(cache.count).to eq 1
+    expect(cache.read(:one)).to eq "1"
+
+    cache.write(:one, "11")
+    expect(cache.count).to eq 1
+    expect(cache.read(:one)).to eq "11"
+
+    cache.write(:two, "2")
+    cache.write(:three, "3")
+    expect(cache.count).to eq 3
+
+    cache.write(:four, "4")
+    expect(cache.count).to eq 3
+
+    expect(cache.hash.keys).to include(:two, :three, :four)
+
+    cache.read(:two)
+    cache.read(:four)
+    cache.write(:five, "5")
+    expect(cache.hash.keys).to include(:two, :four, :five)
```

Comment:
Interestingly, `expect(cache.hash.keys).to match [:five, :two, :four]` fails.

I know that hashes keep the keys in insert order, but nevertheless this had surprised me.